### PR TITLE
Add field methods to Table class

### PR DESCRIPTION
**Pull request by stcheng**

### CODE REVIEW

File: common/table.cpp [hunk 1]
```diff
@@ -83,6 +83,61 @@ void Table::del(std::string key, std::string /* op */)
                            "DEL operation failed");
 }
 
+std::string Table::getField(std::string key, std::string field)
```
Comment (stcheng) [inline common/table.cpp]: This function will be combined with tryGetField.

File: common/table.cpp [hunk 2]
```diff
@@ -83,6 +83,48 @@ void Table::del(std::string key, std::string /* op */)
                            "DEL operation failed");
 }
 
+bool Table::getField(std::string key, std::string field, std::string &value)
+{
+    redisReply *reply = (redisReply*)redisCommand(
```
Comment (stcheng) [inline common/table.cpp]: Is it better to use RedisReply class to do the staffs? Similar to the previous Table::get function

File: common/table.cpp [hunk 3]
```diff
@@ -83,6 +83,48 @@ void Table::del(std::string key, std::string /* op */)
...
+    RedisReply r(m_db, string("HDEL ") + getKeyName(key) + " " + field, REDIS_REPLY_INTEGER);
+    if (r.getContext()->type != REDIS_REPLY_INTEGER)
+        throw system_error(make_error_code(errc::io_error),
+                           "DEL operation failed");
+}
+
+Table::~Table()
+{
+}
+
```
Comment (eladraz) [inline common/table.cpp]: Why do you need empty destructor?

File: common/table.cpp [hunk 4]
```diff
@@ -83,6 +83,48 @@ void Table::del(std::string key, std::string /* op */)
...
+    FieldValueTuple entry(field, value);
+
+    std::vector<FieldValueTuple> values { entry };
+
+    set(key, values);
+}
+
+void Table::delField(std::string key, std::string field)
+{
+    RedisReply r(m_db, string("HDEL ") + getKeyName(key) + " " + field, REDIS_REPLY_INTEGER);
```
Comment (eladraz) [inline common/table.cpp]: Please use redisFormatCommand. (See formatHSet API for exmaple)
'field' can be anything...

File: common/table.cpp [hunk 5]
```diff
@@ -83,6 +83,48 @@ void Table::del(std::string key, std::string /* op */)
                            "DEL operation failed");
 }
 
+bool Table::getField(std::string key, std::string field, std::string &value)
+{
+    redisReply *reply = (redisReply*)redisCommand(
+            m_db->getContext(),
+            "HGET %s %s",
+            getKeyName(key).c_str(),
+            field.c_str());
```
Comment (eladraz) [inline common/table.cpp]: Please use redisFormatCommand. (See formatHSet API for exmaple)
'field' can be anything...
Please use RedisReply class as @stcheng commented.

File: common/table.cpp (hunk 3 above)
Comment (kcudnik) [inline common/table.cpp]: since declared as virtual

**Comment by stcheng** [comment]
Acked-by: Shuotian Cheng <shuche@microsoft.com>

**Pull request merged**